var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -0,0 +1,253 @@
+            # os.environ['ESMF_LAPACK_LIBPATH'] = spec['lapack'].prefix.lib
+
+            # Specifies the linker directive needed to link the LAPACK library
+            # to the application.
+            os.environ['ESMF_LAPACK_LIBS'] = spec['lapack'].lapack_libs.link_flags  # noqa
+        else:
+            # Disables LAPACK-dependent code.
+            os.environ['ESMF_LAPACK'] = 'OFF'
+
+        ##########
+        # NetCDF #
+        ##########
+
+        if '+netcdf' in spec:
+            # ESMF provides the ability to read Grid and Mesh data in
+            # NetCDF format.
+            if spec.satisfies('^netcdf@4.2:'):
+                # ESMF_NETCDF_LIBS will be set to "-lnetcdff -lnetcdf".
+                # This option is useful for systems which have the Fortran
+                # and C bindings archived in seperate library files.
+                os.environ['ESMF_NETCDF'] = 'split'
+            else:
+                # ESMF_NETCDF_LIBS will be set to "-lnetcdf".
+                # This option is useful when the Fortran and C bindings
+                # are archived together in the same library file.
+                os.environ['ESMF_NETCDF'] = 'standard'
+
+            # FIXME: determine whether or not we need to set these.
+            # ESMF_NETCDF_INCLUDE
+            # ESMF_NETCDF_LIBPATH
+
+        ###################
+        # Parallel-NetCDF #
+        ###################
+
+        if '+pnetcdf' in spec:
+            # ESMF provides the ability to write Mesh weights
+            # using Parallel-NetCDF.
+
+            # When defined, enables the use of Parallel-NetCDF.
+            # ESMF_PNETCDF_LIBS will be set to "-lpnetcdf".
+            os.environ['ESMF_PNETCDF'] = 'standard'
+
+            # FIXME: determine whether or not we need to set these.
+            # ESMF_PNETCDF_INCLUDE
+            # ESMF_PNETCDF_LIBPATH
+
+        ##############
+        # ParallelIO #
+        ##############
+
+        if '+pio' in spec and '+mpi' in spec:
+            # ESMF provides the ability to read and write data in both binary
+            # and NetCDF formats through ParallelIO (PIO), a third-party IO
+            # software library that is integrated in the ESMF library.
+
+            # PIO-dependent features will be enabled and will use the
+            # PIO library that is included and built with ESMF.
+            os.environ['ESMF_PIO'] = 'internal'
+        else:
+            # Disables PIO-dependent code.
+            os.environ['ESMF_PIO'] = 'OFF'
+
+        ##########
+        # XERCES #
+        ##########
+
+        if '+xerces' in spec:
+            # ESMF provides the ability to read Attribute data in
+            # XML file format via the XERCES C++ library.
+
+            # ESMF_XERCES_LIBS will be set to "-lxerces-c".
+            os.environ['ESMF_XERCES'] = 'standard'
+
+            # FIXME: determine if the following are needed
+            # ESMF_XERCES_INCLUDE
+            # ESMF_XERCES_LIBPATH
```

Comment:
@adamjstewart: Lack of `RPATH` shouldn't result cause any undefined symbols.  The `R` in `RPATH` stands for `Runtime` -- it is a runtime search path for the dynamic linker.  But the `-L` argument is the compile time search path.  But that's set by the compiler wrapper, too, so it may well be that the Spack wrappers aren't getting injected.

When I looked at this package, it looked like it had a directory full of configuration makefiles for different systems.  What I typically do for builds like this is pick one of those (maybe a generic one or a Linux one) and patch it with lots of `filter_file` commands so that the compiler wrappers are used correctly.  Then I force the build to use the makefile I modified.